x-pack/plugins/siem/server/lib/detection_engine/signals/filter_events_with_list.test.ts

```diff
@@ -0,0 +1,160 @@
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import uuid from 'uuid';
+import { filterEventsAgainstList } from './filter_events_with_list';
+import { mockLogger, repeatedSearchResultsWithSortId } from './__mocks__/es_results';
+
+import { ListClient } from '../../../../../lists/server';
+import { ListItemArraySchema } from '../../../../../lists/common/schemas';
+
+describe('filterEventsAgainstList', () => {
+  const someGuids = Array.from({ length: 13 }).map((x) => uuid.v4());
+
+  it('should respond with eventSearchResult if exceptionList is empty', async () => {
+    const res = await filterEventsAgainstList({
+      logger: mockLogger,
+      listClient: ({
+        getListItemByValues: async () => ([] as unknown) as ListItemArraySchema,
+      } as unknown) as ListClient,
+      exceptionsList: undefined,
+      eventSearchResult: repeatedSearchResultsWithSortId(4, 4, someGuids.slice(0, 3), [
+        '1.1.1.1',
+        '2.2.2.2',
+        '3.3.3.3',
+        '7.7.7.7',
+      ]),
+    });
+    expect(res.hits.hits.length).toEqual(4);
+  });
+
+  it('should throw an error if malformed exception list present', async () => {
+    try {
+      await filterEventsAgainstList({
+        logger: mockLogger,
+        listClient: ({
+          getListItemByValues: async () => ([] as unknown) as ListItemArraySchema,
+        } as unknown) as ListClient,
+        exceptionsList: [
+          {
+            field: 'source.ip',
+            values_operator: 'excluded',
+            values_type: 'list',
+            values: undefined,
+          },
+        ],
+        eventSearchResult: repeatedSearchResultsWithSortId(4, 4, someGuids.slice(0, 3), [
+          '1.1.1.1',
+          '2.2.2.2',
+          '3.3.3.3',
+          '7.7.7.7',
+        ]),
+      });
+    } catch (exc) {
+      expect(exc.message).toContain('Malformed exception list provided');
+    }
+  });
+
+  it('should throw an error if unsupported exception type', async () => {
+    try {
+      const res = await filterEventsAgainstList({
+        logger: mockLogger,
+        listClient: ({
+          getListItemByValues: async () => ([] as unknown) as ListItemArraySchema,
+        } as unknown) as ListClient,
+        exceptionsList: [
+          {
+            field: 'source.ip',
+            values_operator: 'excluded',
+            values_type: 'list',
+            values: [
+              {
+                id: 'ci-badguys.txt',
+                name: 'unsupportedListPluginType',
+              },
+            ],
+          },
+        ],
+        eventSearchResult: repeatedSearchResultsWithSortId(4, 4, someGuids.slice(0, 3), [
+          '1.1.1.1',
+          '2.2.2.2',
+          '3.3.3.3',
+          '7.7.7.7',
+        ]),
+      });
+      expect(res.hits.hits.length).toEqual(4);
+    } catch (exc) {
+      expect(exc.message).toContain('Unsupported list type used, please use one of');
```

Comment:
I think you're doing yourself a disservice using `toContain` rather than `toEqual` where you can use `toEqual`.

When I changed this out for `expect(exc.message).toEqual()` I got this:

```ts
    Expected: "Unsupported list type used, please use one of"
    Received: "Failed to query lists index. Reason: Unsupported list type used, please use one of [object Object]"
```
Looking closer you can see this part:

```ts
please use one of [object Object]
```

Which tells me that maybe your error messaging has a bug and isn't what you want it to be. I will put the suggestion for the bug fix below, but I would consider using `toEqual` where you can to test and ensure that your error messages say what you want them to say.

Reply:
good catch thank you!